src/main/java/io/kabanero/api/InstanceEndpoints.java

```diff
@@ -42,21 +51,35 @@
     @GET
     @Path("/")
     @Produces(MediaType.APPLICATION_JSON)
-    public Collection<KabaneroInstance> getAllInstances() {
-        return KabaneroManager.getKabaneroManagerInstance().getAllKabaneroInstances();
+    public Response getAllInstances() throws IOException, ApiException, GeneralSecurityException {
+        KabaneroList kabaneros = KabaneroClient.getInstances();
+        if (kabaneros == null || kabaneros.getItems().size() == 0 ){
```

Comment:
I would say take away `kabaneros.getItems().size() == 0` and just have an empty array returned. This takes away some of the guesswork from the clients perspective. for example, what does "No instances found" really mean?" - "0 total instances?", "0 instances that I can access?", "some server error that couldn't get any instances", etc...

Then if kabaneros are null we probably should throw a 500 since that is not expected at all.